### ISSUE TITLE
Calendar NumberFormatException with SOGo groupware

### ISSUE THREAD

**Issue by wilbertvolkers** [gggard/AndroidCaldavSyncAdapater#89]
The calendar color that I got from SOGo Groupware was #CCFF99FF, the trailing FF is probably transparency. This results in a NumberFormatException in the Calendar (see stack below). When I use #CCFF99 all is fine. I made the following change to allow the Calendar to work in my situation.

Calendar.java:

``` java
public void setCalendarColor(String color) {        
    int maxlen=6;
    if(color != null && color.length()>0) {
        if(color.substring(0,1).equals("#")) {
            maxlen++;
        }
        if(color.length()>maxlen) {
            color=color.substring(0,maxlen);
        }           
    }       
    calendarColor = color;
}
```

Stacktrace:

D/AndroidRuntime(836): Shutting down VM
W/dalvikvm(836): threadid=1: thread exiting with uncaught exception (group=0x41465700)
E/AndroidRuntime(836): FATAL EXCEPTION: main
**E/AndroidRuntime(836): java.lang.NumberFormatException: Invalid int: "3439303167"**

NB: 3439303167 = hexadecimal: CCFF99FF

E/AndroidRuntime(836):  at java.lang.Integer.invalidInt(Integer.java:138)
E/AndroidRuntime(836):  at java.lang.Integer.parse(Integer.java:378)
E/AndroidRuntime(836):  at java.lang.Integer.parseInt(Integer.java:366)
E/AndroidRuntime(836):  at java.lang.Integer.parseInt(Integer.java:332)
E/AndroidRuntime(836):  at android.database.MatrixCursor.getInt(MatrixCursor.java:247)
**E/AndroidRuntime(836):    at com.android.calendar.widget.CalendarAppWidgetModel.buildFromCursor(CalendarAppWidgetModel.java:292)**

Found in Google code, the color will be read into an int, max 2^31-1, line 292: http://grepcode.com/file/repository.grepcode.com/java/ext/com.google.android/android-apps/4.3_r2.1/com/android/calendar/widget/CalendarAppWidgetModel.java/#292

E/AndroidRuntime(836):  at com.android.calendar.widget.CalendarAppWidgetService$CalendarFactory.buildAppWidgetModel(CalendarAppWidgetService.java:416)
E/AndroidRuntime(836):  at com.android.calendar.widget.CalendarAppWidgetService$CalendarFactory.onLoadComplete(CalendarAppWidgetService.java:498)
E/AndroidRuntime(836):  at com.android.calendar.widget.CalendarAppWidgetService$CalendarFactory.onLoadComplete(CalendarAppWidgetService.java:113)
E/AndroidRuntime(836):  at android.content.Loader.deliverResult(Loader.java:144)
E/AndroidRuntime(836):  at android.content.CursorLoader.deliverResult(CursorLoader.java:110)
E/AndroidRuntime(836):  at android.content.CursorLoader.deliverResult(CursorLoader.java:43)
E/AndroidRuntime(836):  at android.content.AsyncTaskLoader.dispatchOnLoadComplete(AsyncTaskLoader.java:256)
E/AndroidRuntime(836):  at android.content.AsyncTaskLoader$LoadTask.onPostExecute(AsyncTaskLoader.java:91)
E/AndroidRuntime(836):  at android.os.AsyncTask.finish(AsyncTask.java:631)
E/AndroidRuntime(836):  at android.os.AsyncTask.access$600(AsyncTask.java:177)
E/AndroidRuntime(836):  at android.os.AsyncTask$InternalHandler.handleMessage(AsyncTask.java:644)
E/AndroidRuntime(836):  at android.os.Handler.dispatchMessage(Handler.java:99)
E/AndroidRuntime(836):  at android.os.Looper.loop(Looper.java:137)
E/AndroidRuntime(836):  at android.app.ActivityThread.main(ActivityThread.java:5103)
E/AndroidRuntime(836):  at java.lang.reflect.Method.invokeNative(Native Method)
E/AndroidRuntime(836):  at java.lang.reflect.Method.invoke(Method.java:525)
E/AndroidRuntime(836):  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:737)
E/AndroidRuntime(836):  at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:553)
E/AndroidRuntime(836):  at dalvik.system.NativeStart.main(Native Method)

**Comment by timoberger** [comment]
thanks wilbertvolker,
this should be fixed with c21e4af421e2ed64c7b97c22101ba695cab69593

**Comment by wilbertvolkers** [comment]
That was so fast that i was still editing the issue :) Thanks!
